### ISSUE TITLE
Trivy parser only imports vulnerabilities

### ISSUE THREAD

**Issue by bgoareguer** [DefectDojo/django-DefectDojo#6308]
**Bug description**
Trivy reports can contain:
- vulnerabilities
- misconfigurations
- secrets

But the Trivy parser only imports vulnerabilities.

**Expected behavior**
The Trivy parser should also import misconfigurations and secrets.

**Deployment method**
- [ ] Docker Compose
- [x] Kubernetes
- [ ] GoDojo

**Environment information**
 - Operating System: Ubuntu 20.04
 - DefectDojo version: 2.10.0

**Sample scan files**
```
{
  "SchemaVersion": 2,
  "ArtifactName": ".",
  "ArtifactType": "filesystem",
  "Metadata": {
    "ImageConfig": {
      "architecture": "",
      "created": "0001-01-01T00:00:00Z",
      "os": "",
      "rootfs": {
        "type": "",
        "diff_ids": null
      },
      "config": {}
    }
  },
  "Results": [
    {
      "Target": "requirements.txt",
      "Class": "lang-pkgs",
      "Type": "pip",
      "Vulnerabilities": [
        {
          "VulnerabilityID": "CVE-2021-33503",
          "PkgName": "urllib3",
          "InstalledVersion": "1.26.3",
          "FixedVersion": "1.26.5",
          "Layer": {},
          "SeveritySource": "nvd",
          "PrimaryURL": "https://avd.aquasec.com/nvd/cve-2021-33503",
          "DataSource": {
            "ID": "osv",
            "Name": "Python Packaging Advisory Database",
            "URL": "https://github.com/pypa/advisory-db"
          },
          "Title": "python-urllib3: ReDoS in the parsing of authority part of URL",
          "Description": "An issue was discovered in urllib3 before 1.26.5. When provided with a URL containing many @ characters in the authority component, the authority regular expression exhibits catastrophic backtracking, causing a denial of service if a URL were passed as a parameter or redirected to via an HTTP redirect.",
          "Severity": "HIGH",
          "CweIDs": [
            "CWE-400"
          ],
          "CVSS": {
            "nvd": {
              "V2Vector": "AV:N/AC:L/Au:N/C:N/I:N/A:P",
              "V3Vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
              "V2Score": 5,
              "V3Score": 7.5
            },
            "redhat": {
              "V3Vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
              "V3Score": 7.5
            }
          },
          "References": [
            "https://access.redhat.com/security/cve/CVE-2021-33503",
            "https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-33503",
            "https://errata.almalinux.org/8/ALSA-2021-4162.html",
            "https://github.com/advisories/GHSA-q2q7-5pp4-w6pg",
            "https://github.com/urllib3/urllib3/commit/2d4a3fee6de2fa45eb82169361918f759269b4ec",
            "https://github.com/urllib3/urllib3/security/advisories/GHSA-q2q7-5pp4-w6pg",
            "https://linux.oracle.com/cve/CVE-2021-33503.html",
            "https://linux.oracle.com/errata/ELSA-2021-4162.html",
            "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/6SCV7ZNAHS3E6PBFLJGENCDRDRWRZZ6W/",
            "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/FMUGWEAUYGGHTPPXT6YBD53WYXQGVV73/",
            "https://nvd.nist.gov/vuln/detail/CVE-2021-33503",
            "https://security.gentoo.org/glsa/202107-36",
            "https://www.oracle.com/security-alerts/cpuoct2021.html"
          ],
          "PublishedDate": "2021-06-29T11:15:00Z",
          "LastModifiedDate": "2021-12-08T20:34:00Z"
        },
        {
          "VulnerabilityID": "CVE-2021-28363",
          "PkgName": "urllib3",
          "InstalledVersion": "1.26.3",
          "FixedVersion": "1.26.4",
          "Layer": {},
          "SeveritySource": "nvd",
          "PrimaryURL": "https://avd.aquasec.com/nvd/cve-2021-28363",
          "DataSource": {
            "ID": "osv",
            "Name": "Python Packaging Advisory Database",
            "URL": "https://github.com/pypa/advisory-db"
          },
          "Title": "python-urllib3: HTTPS proxy host name not validated when using default SSLContext",
          "Description": "The urllib3 library 1.26.x before 1.26.4 for Python omits SSL certificate validation in some cases involving HTTPS to HTTPS proxies. The initial connection to the HTTPS proxy (if an SSLContext isn't given via proxy_config) doesn't verify the hostname of the certificate. This means certificates for different servers that still validate properly with the default urllib3 SSLContext will be silently accepted.",
          "Severity": "MEDIUM",
          "CweIDs": [
            "CWE-295"
          ],
          "CVSS": {
            "nvd": {
              "V2Vector": "AV:N/AC:L/Au:N/C:P/I:P/A:N",
              "V3Vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:L/I:L/A:N",
              "V2Score": 6.4,
              "V3Score": 6.5
            },
            "redhat": {
              "V3Vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:L/I:L/A:N",
              "V3Score": 6.5
            }
          },
          "References": [
            "https://access.redhat.com/security/cve/CVE-2021-28363",
            "https://github.com/advisories/GHSA-5phf-pp7p-vc2r",
            "https://github.com/pypa/advisory-db/tree/main/vulns/urllib3/PYSEC-2021-59.yaml",
            "https://github.com/urllib3/urllib3/blob/main/CHANGES.rst#1264-2021-03-15",
            "https://github.com/urllib3/urllib3/commit/8d65ea1ecf6e2cdc27d42124e587c1b83a3118b0",
            "https://github.com/urllib3/urllib3/commits/main",
            "https://github.com/urllib3/urllib3/releases/tag/1.26.4",
            "https://github.com/urllib3/urllib3/security/advisories/GHSA-5phf-pp7p-vc2r",
            "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/4S65ZQVZ2ODGB52IC7VJDBUK4M5INCXL/",
            "https://nvd.nist.gov/vuln/detail/CVE-2021-28363",
            "https://pypi.org/project/urllib3/1.26.4/",
            "https://security.gentoo.org/glsa/202107-36",
            "https://www.oracle.com/security-alerts/cpuoct2021.html"
          ],
          "PublishedDate": "2021-03-15T18:15:00Z",
          "LastModifiedDate": "2021-12-08T19:58:00Z"
        }
      ]
    },
    {
      "Target": "Dockerfile",
      "Class": "config",
      "Type": "dockerfile",
      "MisconfSummary": {
        "Successes": 22,
        "Failures": 1,
        "Exceptions": 0
      },
      "Misconfigurations": [
        {
          "Type": "Dockerfile Security Check",
          "ID": "DS002",
          "Title": "Image user should not be 'root'",
          "Description": "Running containers with 'root' user can lead to a container escape situation. It is a best practice to run containers as non-root users, which can be done by adding a 'USER' statement to the Dockerfile.",
          "Message": "Specify at least 1 USER command in Dockerfile with non-root user as argument",
          "Namespace": "builtin.dockerfile.DS002",
          "Query": "data.builtin.dockerfile.DS002.deny",
          "Resolution": "Add 'USER \u003cnon root user name\u003e' line to the Dockerfile",
          "Severity": "HIGH",
          "PrimaryURL": "https://avd.aquasec.com/misconfig/ds002",
          "References": [
            "https://docs.docker.com/develop/develop-images/dockerfile_best-practices/",
            "https://avd.aquasec.com/misconfig/ds002"
          ],
          "Status": "FAIL",
          "Layer": {},
          "CauseMetadata": {
            "Provider": "Dockerfile",
            "Service": "general",
            "Code": {
              "Lines": null
            }
          }
        }
      ]
    },
    {
      "Target": "Dockerfile",
      "Class": "secret",
      "Secrets": [
        {
          "RuleID": "github-pat",
          "Category": "GitHub",
          "Severity": "CRITICAL",
          "Title": "GitHub Personal Access Token",
          "StartLine": 24,
          "EndLine": 24,
          "Match": "ENV GITHUB_PAT=*****"
        }
      ]
    },
    {
      "Target": "secret.txt",
      "Class": "secret",
      "Secrets": [
        {
          "RuleID": "github-pat",
          "Category": "GitHub",
          "Severity": "CRITICAL",
          "Title": "GitHub Personal Access Token",
          "StartLine": 1,
          "EndLine": 1,
          "Match": "GITHUB_PAT=*****"
        }
      ]
    }
  ]
}

```

**Comment by manuel-sommer** [comment]
@bgoareguer, could you provide scan results to improve the parser?

**Comment by bgoareguer** [comment]
I added a more complete one in the isuue description.

**Comment by StefanFl** [comment]
fixed in release 2.11.0
